Rewrote compound types tour #742

mghildiy opened the pull request.

mghildiy commented:
PR 'Rewrote compound types tour #742' can be closed.

gmethvin commented on _tour/self-types.md:
```diff
@@ -24,12 +22,12 @@ trait User {
 trait Tweeter {
-  this: User =>  // reassign this
-  def tweet(tweetText: String) = println(s"$username: $tweetText")
+	this: User =>  // reassign this
```
Is there a reason this was changed from two spaces to a tab? I'd prefer keeping two-space indents.

jvican commented on _tour/traits.md:
```diff
@@ -36,6 +34,12 @@ Extending the `trait Iterator[A]` requires a type `A` and implementations of the
+  def hasNext: Boolean
+  def next(): A
+}
+
```
Could we remove this extra space? :smile:

mghildiy replied:
I assume by extra space you mean line #41.

mghildiy replied:
or extra space in 38,39.

jvican commented on _tour/traits.md:
```diff
@@ -50,13 +54,13 @@ class IntIterator(to: Int) extends Iterator[Int] {
 val iterator = new IntIterator(10)
-println(iterator.next())  // prints 0
-println(iterator.next())  // prints 1
+iterator.next()  // prints 0
```
`// return 0` rather than `// prints 0`?

mghildiy commented:
I have replaced the tab with two spaces.

jvican commented:
@mghildiy If you're looking for more, we still need one more contributor in https://github.com/scala/docs.scala-lang/pull/746 :smile:.

mghildiy commented:
Would pick #746.Any deadline for it?.

On Sun, Jan 14, 2018 at 3:42 PM, Jorge <notifications@github.com> wrote:

> @mghildiy <https://github.com/mghildiy> If you're looking for more, we
> still need one more contributor in #746
> <https://github.com/scala/docs.scala-lang/pull/746> 😄.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scala/docs.scala-lang/pull/988#issuecomment-357501318>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJly1ps1nP_Qofa75nd6G5NXqpICBAtBks5tKdMPgaJpZM4RdVBw>
> .
>

jvican commented:
Absolutely not, take your time. That's one of the gems that can actually be enlightening to work on if you're new to Scala; a solid understanding of implicits is paramount to become a better Scala programmer..

jvican commented:
Let me know when you've addressed these comments and we merge @mghildiy.

mghildiy commented:
I have already made the change(Ttab to 2 spaces in self-types.md) as
suggested by @gmethvin and created the PR.

On Sun, Jan 14, 2018 at 9:53 PM, Jorge <notifications@github.com> wrote:

> Let me know when you've addressed these comments and we merge @mghildiy
> <https://github.com/mghildiy>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scala/docs.scala-lang/pull/988#issuecomment-357523070>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJly1p6kDmW-J7Lrts0AL_mrM1rht45Gks5tKinkgaJpZM4RdVBw>
> .
>

jvican commented:
It seems you forgot to address my review comments though?

mghildiy commented:
Changes done

SethTisue commented:
thank you @mghildiy!
